docs/DeveloperGuide.md

```diff
@@ -1,38 +1,135 @@
+#### Operations
+- `setLimit(double)`: Ensures limits are non-negative.
+- `formatLimit(double)`: Formats the budget limit for display.
+
+### ExpenseTracker
+#### Purpose
+Track and manage expenses and categories.
+#### Operations
+##### Adding Expenses and Categories
+- `addExpense`: Adds an expense after verifying or creating the necessary category.
+- `addCategory`: Adds a new category if it does not exist.
+
+##### Monthly Budget Reset Functionality
+- `toggleAutoReset`: Toggles automatic budget resets on or off.
+- `checkAndResetBudgets`: Checks for a new month and triggers budget resets if enabled.
+- `resetBudgets`: Resets all budgets to their predefined limits.
+
+##### Viewing and Organizing Data
+- `viewExpensesByCategory`: Displays expenses organized by categories.
+- `viewBudget`: Displays budget limits and current expenditures for each category.
+
+##### Modifying Data
+- `deleteExpense`: Deletes an expense based on its index.
+- `tagExpense`: Reassigns an expense to a different category based on user input.
+
+### Duke
+#### Purpose
+Interface for command-line interactions.
+#### Main flow
+- The system starts and displays a greeting.
+- Continuously processes user commands until "bye".
+- Directly invokes methods from ExpenseTracker based on input.
+
+## Product Scope
+<!-- @@author glenda-1506 -->
+### Target User Profile
+SpendSwift is designed for budget-conscious individuals who prefer a simple, efficient way to track expenses and manage budgets using text commands.
+
+### Value Proposition
+SpendSwift provides a fast, text-based solution for managing finances, eliminating the complexity of traditional budgeting tools.
 
 ## User Stories
-
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+<!-- @@author glenda-1506 -->
+| Version | As a... | I want to... | So that I can... |
+|---------|---------|--------------|------------------|
+| v1.0    | Budget-conscious user | Quickly log an expense using a typed command (e.g., add 50 groceries) | track my spending with easy input |
+| v1.0    | Budget-conscious user | View my budget for all categories | see how much I could spend |
+| v1.0    | Budget-conscious user | Set a monthly reset for my budget tracking | start each month fresh with my budgeting |
+| v1.0    | Budget planner | View all my expenses | monitor what I have been spending on |
+| v1.0    | Frequent user | Set a budget limit for each category (e.g., set budget 200 groceries) | limit my spending according to categories |
+| v1.0    | Frequent user | Delete an expense entry (e.g., delete 5) | quickly correct mistakes |
+| v1.0    | Frequent user | Categorize expenses (e.g., add category food) | customize my expense tracking to better manage my budget |
```

Comment:
Sorry, this is difficult to view in the html site. I have seen the User Stories here and I think it works, but you may want to check on the formatting.
![image](https://github.com/user-attachments/assets/4e0819b4-842a-419e-9415-aa725670b50c)